src/emqx_protocol.erl

```diff
@@ -441,6 +532,21 @@ connack({ReasonCode, PState = #pstate{proto_ver = ProtoVer}}) ->
     _ = deliver({connack, ReasonCode1}, PState),
     {error, emqx_reason_codes:name(ReasonCode1, ProtoVer), PState}.
 
+%%------------------------------------------------------------------------------
+%% Disconnect --> Client
+%%------------------------------------------------------------------------------
+
+disconnect({?RC_SUCCESS, PState}) ->
+    deliver({disconnect, ?RC_SUCCESS}, PState);
+disconnect({ReasonCode, PState = #pstate{proto_ver = ProtoVer}}) ->
+    ReasonCode1 = if ProtoVer =:= ?MQTT_PROTO_V5 ->
+                         ReasonCode;
+                     true ->
```

Comment:
align with `if`.
and a nitpicking: I personally find case clauses much more comfortable to read, so I never write `if`s

Reply:
So am i.